packages/realm_dart/lib/src/configuration.dart

```diff
@@ -398,21 +398,31 @@ class InMemoryConfiguration extends Configuration {
 /// A collection of properties describing the underlying schema of a [RealmObjectBase].
 ///
 /// {@category Configuration}
-class SchemaObject {
+class SchemaObject extends Iterable<SchemaProperty> {
+  final List<SchemaProperty> _properties;
+
   /// Schema object type.
   final Type type;
 
-  /// Collection of the properties of this schema object.
-  final List<SchemaProperty> properties;
-
   /// Returns the name of this schema type.
   final String name;
 
   /// Returns the base type of this schema object.
   final ObjectType baseType;
 
   /// Creates schema instance with object type and collection of object's properties.
-  const SchemaObject(this.baseType, this.type, this.name, this.properties);
+  SchemaObject(this.baseType, this.type, this.name, Iterable<SchemaProperty> properties) : _properties = List.from(properties);
```

Comment:
Of the top of my head, I think you can use a [DelegatingIterable](https://api.flutter.dev/flutter/package-collection_collection/DelegatingIterable/DelegatingIterable.html) .. or just drop making `SchemaObject` iterable itself?

Reply:
The problem is that we want the `_properties` list to be mutable - i.e. we're adding stuff at runtime. My understanding was that this is incompatible with const-constructed objects, but it's very possible I'm missing something.

Reply:
Just because a ctor is marked as `const` doesn't mean it has to be used with `const` arguments. Only that it **can**. If the backing list to `DelegatingIterable` is const, then you will get a runtime exception trying to modify it.

I guess that ties into my question, if we should track both the _current_ schema, and the _original_ schema used when generating the model. The latter could still be `const` constructed.

Reply:
I still dream of using the flyweight property pattern from the old #903 PR 😄